test/framework/fileConversion.go

```diff
@@ -37,44 +40,86 @@ func FormatTestData(srcFile, tgtDir string, targetFormats ...string) (string, er
 	return srcFile, nil
 }
 
-func tarCmd(src, tgtDir string) (string, error) {
-	base := filepath.Base(src)
-	tgt := filepath.Join(tgtDir, base+image.ExtTar)
-	args := []string{"-cf", tgt, src}
+func toTar(src, tgtDir string) (string, error) {
+	tgtFile, tgtPath, err := createTargetFile(src, tgtDir, image.ExtTar)
+	defer tgtFile.Close()
 
-	if err := doCmdAndVerifyFile(tgt, "tar", args...); err != nil {
-		return "", err
+	w := tar.NewWriter(tgtFile)
+	defer w.Close()
+
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return "", errors.Wrap(err, "Error opening file")
 	}
-	return tgt, nil
+	defer srcFile.Close()
+
+	srcFileInfo, err := srcFile.Stat()
+	if err != nil {
+		return "", errors.Wrap(err, "Error stating file")
+	}
+
+	hdr, err := tar.FileInfoHeader(srcFileInfo, "")
+	if err != nil {
+		return "", errors.Wrap(err, "Error generating tar file header")
+	}
+
+	err = w.WriteHeader(hdr)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing header")
+	}
+	defer w.Close()
+
+	_, err = io.Copy(w, srcFile)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing to file")
+	}
+	return tgtPath, nil
 }
 
-func gzCmd(src, tgtDir string) (string, error) {
-	src, err := copyIfNotPresent(src, tgtDir)
+func toGz(src, tgtDir string) (string, error) {
+	tgtFile, tgtPath, err := createTargetFile(src, tgtDir, image.ExtGz)
+	defer tgtFile.Close()
+
+	w := gzip.NewWriter(tgtFile)
+	defer w.Close()
+
+	srcFile, err := os.Open(src)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "Error opening file")
 	}
-	base := filepath.Base(src)
-	tgt := filepath.Join(tgtDir, base+image.ExtGz)
-	if err := doCmdAndVerifyFile(tgt, "gzip", src); err != nil {
-		return "", err
+	defer srcFile.Close()
+
+	_, err = io.Copy(w, srcFile)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing to file")
 	}
-	return tgt, nil
+	return tgtPath, nil
 }
 
-func xzCmd(src, tgtDir string) (string, error) {
-	src, err := copyIfNotPresent(src, tgtDir)
+func toXz(src, tgtDir string) (string, error) {
+	tgtFile, tgtPath, err := createTargetFile(src, tgtDir, image.ExtXz)
+	defer tgtFile.Close()
+
+	w, err := xz.NewWriter(tgtFile)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "Error getting writer")
 	}
-	base := filepath.Base(src)
-	tgt := filepath.Join(tgtDir, base+image.ExtXz)
-	if err := doCmdAndVerifyFile(tgt, "xz", src); err != nil {
-		return "", err
+	defer w.Close()
+
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return "", errors.Wrap(err, "Error opening file")
 	}
-	return tgt, nil
+	defer srcFile.Close()
+
+	_, err = io.Copy(w, srcFile)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing to file")
+	}
+	return tgtPath, nil
 }
 
-func qcow2Cmd(srcfile, tgtDir string) (string, error) {
+func toQcow2(srcfile, tgtDir string) (string, error) {
```

Comment:
`Replace` below doesn't work if _.iso_ is part of base name (which is very unlikely but could occur).